### ISSUE TITLE
Merge states instead of overriding it.

### ISSUE THREAD

**Issue by mxchange** [btroncone/ngrx-store-localstorage#80]
# Current behaviour
When I pass something like:

```ts
{
      keys: [
        {
          app: ['locale']
        }
      ],
      rehydrate: true
    }
```

It will override all my initial `app` state keys, and only fills in the `locale` field.

This will break my app, and forces me to choose to either sync the whole`app` key, or nothing at all. Or alternatively, make an entirely new `locale` reducer on the root.

# Expected behaviour
The state being merged similar to how it happens in a reducer:

```ts
{
   [currentKey]: {
      ...initialState[currentKey],
      ...localStorage[currentKey]
   } 
}
```

**Comment by aprotsyk** [comment]
+1

**Comment by mxchange** [comment]
Anyone find a more active Fork of this repo? It's pretty nice, but last update 3 months ago...

**Comment by btroncone** [comment]
@mxchange We just released new version a few days ago. @tanyagray has done some awesome work already creating and reviewing PRs, hopefully we can be more responsive moving forward! 👍🏻

**Comment by tanyagray** [comment]
This is definitely a good idea, if there is a PR for it I'd be happy to take a look. Need to solve this for my own project too.

I think this is one which would benefit from including a couple of basic tests to support it.

Happy to collaborate on making this happen :)

**Comment by pedep** [comment]
Not sure if there is an "easy" way to have access to the initial state of all reducers, but adding an `initialState` object to `LocalStorageConfig`, and then merging this object with `state` and `rehydratedState` in the initial action might be the easiest until a better solution is found.
Not sure if this solution would be satisfactory, but it solved the issues i were having.

**Comment by BBlackwo** [comment]
Closing this issue as it was fixed in version 6.0.0 which started doing a deep merge of the initial state and the local storage state.
